src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -96,6 +98,46 @@ public EvaluationContext getEvaluationContext() {
         }
     }
 
+    /**
+     * Return the transaction context propagator.
+     */
+    public TransactionContextPropagator getTransactionContextPropagator() {
+        try (AutoCloseableLock __ = lock.readLockAutoCloseable()) {
+            return this.transactionContextPropagator;
+        }
+    }
+
+    /**
+     * Sets the transaction context propagator.
+     *
+     * @throws IllegalArgumentException if {@code transactionContextPropagator} is null
+     */
+    public void setTransactionContextPropagator(TransactionContextPropagator transactionContextPropagator) {
+        if (transactionContextPropagator == null) {
+            throw new IllegalArgumentException("Transaction context propagator cannot be null");
+        }
+        try (AutoCloseableLock __ = lock.writeLockAutoCloseable()) {
+            this.transactionContextPropagator = transactionContextPropagator;
+        }
+    }
+
+    /**
+     * Returns the currently defined transaction context using the registered transaction
+     * context propagator.
+     *
+     * @return {@link EvaluationContext} The current transaction context
+     */
+    public EvaluationContext getTransactionContext() {
```

Comment:
I think this should be package private and avoid public access 

```suggestion
    EvaluationContext getTransactionContext() {
```